packages/esm-patient-chart-app/src/visit/visit-form/visit-form.workspace.tsx

```diff
@@ -299,12 +302,15 @@ const VisitForm: React.FC<VisitFormProps> = ({
             // 1. Current visit data (for critical components like visit summary, action buttons)
             // 2. Visit history table (for the paginated visit list)
 
-            // Update current visit data for critical components (useVisit hook)
-            mutateCurrentVisit();
+            // Update patient's visit data for critical components
+            const mutateSavedOrUpdatedVisit = () => invalidateVisitByUuid(globalMutate, visit.uuid);
+            mutateActiveVisit();
+            setVisitContext?.(visit, mutateSavedOrUpdatedVisit);
```

Comment:
```suggestion
            if (visit?.patient?.uuid === patientUuid) {
              setVisitContext?.(visit, mutateSavedOrUpdatedVisit);
            }
```

Might we want to add this guard against stale context updates?

Reply:
`usePatientChartStore` should already be guarding against he case when the patient we start the visit for is not the same as the patient in the store. In that case, the `setVisitContext` is just a dummy function that does nothing.

That said, I'm slowing leaning towards the idea of having 2 versions of the visit form workspace (albeit they still share a lot of code), one meant for use within the patient chart and one outside. Same with all other workspaces in the patient chart that are used outside, of which there are surprisingly many (at least 12).